src/v/cluster/errc.h

```diff
@@ -59,6 +59,7 @@ enum class errc : int16_t {
     error_collecting_health_report,
     leadership_changed,
     feature_disabled,
+    quota_exceeded,
```

Comment:
I don't see `quota_exceeded` being returned anywhere?  It's referenced in the path for converting to a kafka exception, but I don't see any code that actually originates an error of this type.